src/xrpld/app/tx/detail/PayChan.cpp

```diff
@@ -453,16 +454,33 @@ PayChanClaim::preflight(PreflightContext const& ctx)
             return temBAD_SIGNATURE;
     }
 
+    if (auto const err = credentials::check(ctx); !isTesSuccess(err))
+        return err;
+
     return preflight2(ctx);
 }
 
+TER
+PayChanClaim::preclaim(PreclaimContext const& ctx)
+{
```

Comment:
Should we make it amendment guarded and return tesSUCCESS if the credentials is not enabled?

Reply:
Amendment checks are in preflight

Reply:
Checks added